### PR TITLE
tbi(ui5-config): add authType option and comments to yaml config

### PR DESCRIPTION
**Pull request by cianmSAP**
#1965
 
- (`ui5-config`): Adds new property `authenticationType` to `FioriToolsProxyConfigBackend` interface
- (`ui5-config`): Adds function `getBackendComments` to retrieve comments for backend values
- (`ui5-config`): Updates `addBackendToFioriToolsProxydMiddleware` to check provided backend value for applicable comments and add them to the config

- (`yaml`): adds method `createNode` which creates a `Node` from the provided value and adds any relevant comments to the values
- (`yaml`): updates `toPathArray` to convert index values passed as strings to numbers, to be enable adding or updating config items within an array

### CODE REVIEW

**Review by tobiasqueck — DISMISSED** [review]
* changeset good
* code is clear and easy to understand (despite the fact that it touches the imho complicated yaml handling)
* did not test locally but reviewed the snapshots that show that the code works correctly

**Review by korotkovao — APPROVED** [review]
👍 LGTM, Thanks @cianmSAP

File: packages/ui5-config/src/types/middlewares.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export interface FioriToolsProxyConfigBackend {
     path?: string;
     url: string;
     client?: string;
+    authenticationType?: string;
```
Comment (Klaus-Keller) [inline packages/ui5-config/src/types/middlewares.ts:19]: @cianmSAP, shouldn't this be mapped to type
https://github.com/SAP/open-ux-tools/blob/96de8f25054b1a88da17398fb9a4ad2cc0043752/packages/store/src/entities/backend-system.ts#L4-L9

**Review by zdravko-georgiev — DISMISSED** [review]
Change looks good
Very good test coverage
Did NOT tested manually

**Review by devinea — DISMISSED** [review]
Code changes are clear. 
Changeset ✅

File: packages/ui5-config/src/middlewares.ts [hunk 2]
```diff
@@ -25,6 +25,29 @@ export function getAppReloadMiddlewareConfig(): CustomMiddleware<FioriAppReloadC
...
+ * @param index - optional index of backend entry
+ * @returns the node comments for the backend config
+ */
+export function getBackendComments(
+    backend: FioriToolsProxyConfigBackend,
+    index?: number
+): NodeComment<CustomMiddleware<FioriToolsProxyConfig>>[] {
+    const comment = [];
+
+    if (backend.authenticationType === 'reentranceTicket') {
```
Comment (IainSAP) [inline packages/ui5-config/src/middlewares.ts:41]:
```suggestion
    if (backend.authenticationType === AuthenticationType.ReentranceTicket) {
```

Comment (cianmSAP) [reply packages/ui5-config/src/middlewares.ts]: Updated

File: packages/ui5-config/src/types/middlewares.ts (hunk 1 above)
Comment (cianmSAP) [reply packages/ui5-config/src/types/middlewares.ts]: I've updated this, thanks @Klaus-Keller

**Review by Klaus-Keller — APPROVED** [review]
Thanks @cianmSAP, type reuse of `AuthenticationType` looks good now.

**Review by korotkovao — APPROVED** [review]
Reviews addressed, 👍

File: packages/ui5-config/src/middlewares.ts (hunk 2 above)
Comment (IainSAP) [reply packages/ui5-config/src/middlewares.ts]: Thanks

**Review by devinea — APPROVED** [review]
Re-approving

**Pull request merged**